src/imgcodecs.rs

```diff
@@ -2,111 +2,120 @@
 //! imgcodecs](http://docs.opencv.org/3.1.0/d4/da8/group__imgcodecs.html).
 
 use core::*;
+use errors::*;
 use std::ffi::CString;
-use std::os::raw::{c_char, c_int};
+use std::os::raw::c_char;
 use std::path::Path;
 use failure::Error;
 
 // =============================================================================
 //  Imgproc
 // =============================================================================
 /// ImreadModes
-#[derive(Debug, PartialEq, Clone, Copy)]
-pub enum ImreadModes {
+#[repr(C)]
```

Comment:
While I like the removal of `Imread` prefix in the enum fields, I have concern about this readability improvement. I think it has differed from OpenCV too much.

One way to get the best of both worlds is to improve the documentation so that users know that this ImageReadMode is ImreadModes in OpenCV. Perhaps fix the documentation with a link: https://docs.opencv.org/trunk/d4/da8/group__imgcodecs.html#ga61d9b0126a3e57d9277ac48327799c80

Reply:
Ok. Say when you're done (or is it your only concern in this changeset?)

Reply:
I am done with the review.